crates/shared/src/order_validation.rs

```diff
@@ -330,6 +342,16 @@ impl OrderValidating for OrderValidator {
         if order.sell_token == BUY_ETH_ADDRESS {
             return Err(PartialValidationError::InvalidNativeSellToken);
         }
+        if !self.enable_limit_orders && order.buy_token == BUY_ETH_ADDRESS {
```

Comment:
This should use `enable_smart_contract_payments`, right?
```suggestion
        if !self.enable_smart_contract_payments && order.buy_token == BUY_ETH_ADDRESS {
```
Also if the problem is only paying `ETH` to smart contracts and `ERC20` would be fine (not sure about that), then `enable_smart_contract_payments` sounds a bit too general.